### PR TITLE
Fetch potential matches from database

### PR DESCRIPTION
**Pull request by Lollipoke**
I used Kilian's matching algorithm to fetch potential matches directly from the database instead of having a hardcoded list of profile.
I also fixed the type of ageRange in User.kt

### CODE REVIEW

**Review by cgntnr — APPROVED** [review]
I couldn't notice anything to add, good job. Though we don't have the means to test via Emulator etc. I'm guessing you checked their functioning either by hand or with local tests.

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchingAlgorithm.kt [hunk 1]
```diff
@@ -49,23 +47,19 @@ class MatchingAlgorithm {
...
-            query.get().addOnSuccessListener { users ->
-                for (user in users) {
+            //Wait on the query to be done before continuing
+            val users = query.get().await()
+            for (user in users) {
+                if (user.id != userid) {
                     matches += user.toUser()
                 }
-            }.addOnFailureListener { exception ->
-                Log.w(TAG, "Error getting users : ", exception)
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/match/MatchingAlgorithm.kt:65]: What happens now if the query fails ?

**Review by Xelowak — APPROVED** [review]
Everything looks fine to me.

File: app/src/main/java/ch/epfl/sdp/blindly/match/MatchingAlgorithm.kt (hunk 1 above)
Comment (Lollipoke) [reply app/src/main/java/ch/epfl/sdp/blindly/match/MatchingAlgorithm.kt]: You're right, I added a try-catch :)

**Pull request merged**